src/main/java/me/arminb/sara/dao/QuestionDAO.java

```diff
@@ -0,0 +1,15 @@
+package me.arminb.sara.dao;
+
+import me.arminb.sara.entities.Question;
+
+import java.util.List;
+
+public interface QuestionDAO{
+
+        public List<Question> findAll(Integer pageNumber, Integer pageCount) throws DataAccessException;
```

Comment:
this method seems to be unnecessary cuz find with null parameters will do the same. I would remove it from this interface and replace it everywhere with the find method with null parameters. I would do the same to the UserDAO